keymanapp/keyman
# fix(linux): add `keymanFacename` to .ldml file

This change adds the `keymanFacename` attribute based on UnicodeFont or AnsiFont to the root element of the generated .ldml file so that onboard-keyman can display the proper font for the osk.

Fixes: #12019

# User Testing

## Preparations

- The tests should be run with X11

- disable Universe repo:
  ```
  sudo add-apt-repository --remove universe
  ```
- Add the Keyman Test ppa repo by running:
  ```
  sudo add-apt-repository ppa:keymanapp/keyman-test
  ```
- Install/update `onboard-keyman`
- [Install build artifacts of this PR](https://github.com/keymanapp/keyman/wiki/How-to-test-artifacts-from-pull-requests-for-Keyman-for-Linux)
- Install Khmer Angkor keyboard, version 1.5+
- Install at least one other Keyman keyboard
- Reboot

## Tests

**TEST_MSGDLG_MISSING_FONTTOOLS**:

- with Universe disabled and python3-fonttools not installed, run `km-config`
- verify that you get a dialog telling you that python3-fonttools have to be installed

**TEST_MSGCMDLINE_MISSING_FONTTOOLS**:

- with Universe disabled and python3-fonttools not installed, run `km-config -i khmer_angkor`
- verify that km-config aborts, showing an error that python3-fonttools have to be installed

**TEST_NO_ERROR_WITH_FONTTOOLS**: 

- enable Universe repo:
  ```
  sudo add-apt-repository universe
  ```
- re-install the build artifacts of this PR
- verify that you can open `km-config` without getting an error

**TEST_OSKFONT_SHOWS**: 

- open `onboard`
- switch to Khmer Angkor keyboard
- verify that the key caps on the OSK properly show the expected keys

**TEST_FONT_OFF**:

- switch back to a non-Keyman keyboard
- verify that the key caps properly switch back to English (or whatever)

**TEST_NO_OSKFONT**:

- switch to the other installed Keyman keyboards
- verify that the key caps properly show for each of the keyboards
